packages/flutter/lib/src/services/binary_messenger.dart

```diff
@@ -56,99 +56,21 @@ abstract class BinaryMessenger {
-    final Completer<ByteData> completer = Completer<ByteData>();
-    // ui.window is accessed directly instead of using ServicesBinding.instance.window
-    // because this method might be invoked before any binding is initialized.
-    // This issue was reported in #27541. It is not ideal to statically access
-    // ui.window because the Window may be dependency injected elsewhere with
-    // a different instance. However, static access at this location seems to be
-    // the least bad option.
-    ui.window.sendPlatformMessage(channel, message, (ByteData reply) {
-      try {
-        completer.complete(reply);
-      } catch (exception, stack) {
-        FlutterError.reportError(FlutterErrorDetails(
-          exception: exception,
-          stack: stack,
-          library: 'services library',
-          context: ErrorDescription('during a platform message response callback'),
-        ));
-      }
-    });
-    return completer.future;
-  }
-
-  @override
-  Future<void> handlePlatformMessage(
-    String channel,
-    ByteData data,
-    ui.PlatformMessageResponseCallback callback,
-  ) async {
-    ByteData response;
-    try {
-      final MessageHandler handler = _handlers[channel];
-      if (handler != null)
-        response = await handler(data);
-    } catch (exception, stack) {
-      FlutterError.reportError(FlutterErrorDetails(
-        exception: exception,
-        stack: stack,
-        library: 'services library',
-        context: ErrorDescription('during a platform message callback'),
-      ));
-    } finally {
-      callback(response);
-    }
-  }
-
-
-  @override
-  Future<ByteData> send(String channel, ByteData message) {
-    final MessageHandler handler = _mockHandlers[channel];
-    if (handler != null)
-      return handler(message);
-    return _sendPlatformMessage(channel, message);
-  }
-
-  @override
-  void setMessageHandler(String channel, MessageHandler handler) {
-    if (handler == null)
-      _handlers.remove(channel);
-    else
-      _handlers[channel] = handler;
-  }
-
-  @override
-  void setMockMessageHandler(String channel, MessageHandler handler) {
-    if (handler == null)
-      _mockHandlers.remove(channel);
-    else
-      _mockHandlers[channel] = handler;
-  }
-}
-
 /// The default instance of [BinaryMessenger].
 ///
+/// This API has been deprecated in favor of [ServicesBinding.defaultBinaryMessenger].
+/// Please use [ServicesBinding.defaultBinaryMessenger] as the default
+/// instance of [BinaryMessenger].
+///
 /// This is used to send messages from the application to the platform, and
 /// keeps track of which handlers have been registered on each channel so
 /// it may dispatch incoming messages to the registered handler.
-const BinaryMessenger defaultBinaryMessenger = _DefaultBinaryMessenger._();
+@Deprecated('Use ServicesBinding.instance.defaultBinaryMessenger instead.')
+BinaryMessenger get defaultBinaryMessenger {
+  const String errMsg =
```

Comment:
Nit: just inline the error message into the assert.

I'd also update the text slightly, and we should put this in a `FlutterError` so the associated error message doesn't say that the error implies something is wrong with Flutter.  Putting it all together, it'll look like:

```dart
  assert(() {
    if (ServicesBinding.instance == null) {
      throw FlutterError(
        'ServicesBinding.defaultBinaryMessenger was accessed before the '
        'binding was initialized.\n'
        'If you\'re running an application and need to access the binary '
        'messenger before `runApp()` has been called (for example, during '
        'plugin initialization), then you need to explicitly call the '
        '`WidgetsFlutterBinding.ensureInitialized()` first.\n'
        'If you\'re running a test, you can call the '
        '`TestWidgetsFlutterBinding.ensureInitialized()` as the first line in '
        'your test\'s `main()` method to initialize the binding.'
      );
    }
    return true;
  }());
```

Reply:
Thx! Done.